Enums cause type-punning issue

This XML structure:

```xml
<packet ePacketType = "PAKCS_EQUIP_ITEM_RIDE">
    <includeXml path = "equipped_position_ride.xml" />
    
    <element name = "slot" type = "EquippedPositionRide" />
    <element name = "index" type = "uint16_t" />
</packet>
```

```xml
<simpleType name = "EquippedPositionRide">
    <restriction base = "uint16_t">
        <enumeration value = "BODY" />
        <enumeration value = "ENGINE" />
        <enumeration value = "LEG" />
        <enumeration value = "ABILITY" />
        <enumeration value = "ARM" />
    </restriction>
</simpleType>
```

Ends up causing an error on GCC8:

```
gen/packets/cli_equip_item_ride.cpp: In constructor ‘RoseCommon::Packet::CliEquipItemRide::CliEquipItemRide(RoseCommon::CRoseReader)’:
gen/packets/cli_equip_item_ride.cpp:11:41: error: dereferencing type-punned pointer will break strict-aliasing rules [-Werror=strict-aliasing]
     if (!reader.get_uint16_t((uint16_t&)slot)) {
```

The source of the issue is these lines of code:
```cpp
CliEquipItemRide::CliEquipItemRide(CRoseReader reader) : CRosePacket(reader) {
    if (!reader.get_uint16_t((uint16_t&)slot)) {
        return;
    }
    if (!reader.get_uint16_t(index)) {
        return;
    }
}
```

--- a/src/codegen_source.rs
+++ b/src/codegen_source.rs
@@ -97,11 +97,9 @@ impl<'a, W: Write> CodeSourceGenerator<'a, W> {
                     } else {
                         clean_base(elem.type_())
                     };
-                    let name = if let Some(ref enum_type) = elem.enum_type() {
-                        format!("({}&){}", enum_type, elem.name())
-                    } else {
-                        elem.name().to_owned()
-                    };
+
+                    let name = elem.name().to_owned();
+
                     if let Some(ref o) = elem.occurs() {
                         use ::flat_ast::Occurs::*;
                         match o {
@@ -153,6 +151,15 @@ impl<'a, W: Write> CodeSourceGenerator<'a, W> {
                                 cg!(self, "}}");
                             }
                         }
+                    } else if let Some(ref enum_type) = elem.enum_type() {
+                        // Introduce a temprorary variable to avoid reference aliasing issues
+                        let temp_name = format!("{}_temp", elem.name());
+                        cg!(self, "{} {};", enum_type, temp_name);
+                        self.write_if_else(&format!("!reader.get_{}({})", enum_type, temp_name), &[
+                            "return;"
+                        ], None)?;
+                        cg!(self, "{} = static_cast<{}>({});", elem.name(), elem.type_(), temp_name);
+                        cg!(self);
                     } else {
                         self.write_if_else(&format!("!reader.get_{}({})", base, name), &[
                             "return;"

